contracts/governance/InterchainGovernance.sol

```diff
@@ -18,48 +17,67 @@ contract InterchainGovernance is AxelarExecutable, TimeLock, IInterchainGovernan
         CancelTimeLockProposal
     }
 
+    string public governanceChain;
+    string public governanceAddress;
     bytes32 public immutable governanceChainHash;
     bytes32 public immutable governanceAddressHash;
 
     /**
-     * @dev Initializes the contract
-     * @param gatewayAddress The address of the Axelar gateway contract
+     * @notice Initializes the contract
+     * @param gateway The address of the Axelar gateway contract
      * @param governanceChain_ The name of the governance chain
      * @param governanceAddress_ The address of the governance contract
      * @param minimumTimeDelay The minimum time delay for timelock operations
      */
-    // solhint-disable-next-line no-empty-blocks
     constructor(
-        address gatewayAddress,
+        address gateway,
         string memory governanceChain_,
         string memory governanceAddress_,
         uint256 minimumTimeDelay
-    ) AxelarExecutable(gatewayAddress) TimeLock(minimumTimeDelay) {
+    ) AxelarExecutable(gateway) TimeLock(minimumTimeDelay) {
+        governanceChain = governanceChain_;
+        governanceAddress = governanceAddress_;
         governanceChainHash = keccak256(bytes(governanceChain_));
         governanceAddressHash = keccak256(bytes(governanceAddress_));
     }
 
     /**
-     * @dev Executes a proposal
+     * @notice Returns the ETA of a proposal
+     * @param proposalHash The hash of the proposal
+     * @return uint256 The ETA of the proposal
+     */
+    function getProposalEta(bytes32 proposalHash) external view returns (uint256) {
```

Comment:
what i meant was that this getter takes in the same params as executeProposal, and provides an eta for it, vs passing the hash